### PR TITLE
Clean up some univariate constructors

### PR DESCRIPTION
**Pull request by mgkurtz**
Taken from #1360 as this is now totally orthogonal to the remaining code there.

### CODE REVIEW

File: src/LaurentSeries.jl [hunk 1]
```diff
@@ -24,9 +24,11 @@ object `S` will be cached so that supplying the same base ring, string and
 precision in future will return the same parent object and generator. If
 caching of the parent object is not required, `cached` can be set to `false`.
 """
+
+@doc _laurent_series_ring_doc
```
Comment (fingolfin) [inline src/LaurentSeries.jl:28]: Why do we need to attach this docstring to two methods?

File: src/generic/NumberField.jl [hunk 2]
```diff
@@ -36,7 +38,7 @@ end
 #
 ###############################################################################
 
-function number_field(a::Poly{Rational{BigInt}}, s::Symbol, t = "\$"; cached::Bool=true)
+function number_field(a::Poly{Rational{BigInt}}, ::VarName; cached::Bool=true)
```
Comment (fingolfin) [inline src/generic/NumberField.jl:41]: So no code ever tries to pass `t`?

**Review by fingolfin — COMMENTED** [review]
Some doctests fail due to method not found error :-(

File: src/generic/NumberField.jl (hunk 2 above)
Comment (mgkurtz) [reply src/generic/NumberField.jl:41]: Right, and as of #1485 this code also should not be used anywhere else.

File: src/LaurentSeries.jl (hunk 1 above)
Comment (mgkurtz) [reply src/LaurentSeries.jl:28]: Well, I can of course also let one refer to the other. I don’t care. Currently, the docstring is only attached to `laurent_series_ring`. Attaching it two both seemed to be an improvement.

Comment (mgkurtz) [reply src/LaurentSeries.jl]: alright

**Review by fingolfin — APPROVED** [review]

**Pull request merged**